src/js/controller/instream-adapter.js

```diff
@@ -192,10 +195,33 @@ const InstreamAdapter = function(_controller, _model, _view, _mediaPool) {
         }
     }
 
+    function _triggerAdPlayPause(event) {
+        const { newstate } = event;
+        const adEvent = Object.assign({
+            oldstate: _adProgram.model.get('state')
+        }, _data, event);
+
+        if (adEvent.oldstate === newstate) {
+            return;
+        }
+
+        if (newstate === STATE_PLAYING) {
+            _controller.trigger(AD_PLAY, adEvent);
+            _data = { }; // Reset.
+        } else if (newstate === STATE_PAUSED) {
+            _controller.trigger(AD_PAUSE, adEvent);
+            _data = { }; // Reset.
+        }
+    }
+
+    this.setEventData = function(data) {
+        _data = data;
```

Comment:
This should still be settable (or reset) via `instream.play(meta)`, `instream.pause(meta)` and potentially `instream.loadItem`. Resetting on `init` and `destroy` would be good practice as well.

And what about `_instreamItemNext`? Shouldn't this also reset the ad event data (each ad pod should have a new event data)?